### PR TITLE
SMAPI Wrapper Events + Enum fixes + Patch additions + More specific NullRef + HarmonyBundle fixes.

### PR DESCRIPTION
**Pull request by M3ales**
May want to make better comments on enum, but implementation is fixed.

### CODE REVIEW

**Comment by M3ales** [comment]
I can also generalise NPCUtil's stuff to encompass Character (incl FarmAnimals and other players) with a generic type specifier. See https://github.com/M3ales/RelationshipTooltips/blob/5569cd95595144489f3654acf0b0066638907843/RelationshipTooltips/RelationshipTooltipsMod.cs#L98

**Comment by M3ales** [comment]
Merged changes made, unsure about having Priority in same file as EventBus, but yeah it doesn't really matter. The file fluff of having them separate is minor, and I don't think it's confusing.

File: Bookcase/Events/BookcaseEvents.cs [hunk 1]
```diff
@@ -38,5 +38,27 @@ public static class BookcaseEvents {
...
+        /// Wrapper of SMAPI's GameEvents.HalfSecondTick - fired every 500ms.
+        /// </summary>
+        public static EventBus<Event> GameHalfSecondTick = new EventBus<Event>();
+
+        /// <summary>
+        /// Wrapper of SMAPI's GameEvents.SecondTick - fired every 1000ms.
+        /// </summary>
+        public static EventBus<Event> GameFullSecondTick = new EventBus<Event>();
+
+        public static EventBus<CollectionsPageDrawEvent> CollectionsPageDrawEvent = new EventBus<CollectionsPageDrawEvent>();
```
Comment (darkhax) [inline Bookcase/Events/BookcaseEvents.cs]: This one needs a comment

File: Bookcase/Patches/StardewValley/Menus/IClickableMenu/CollectionsPageDrawPatch.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        public Type TargetType => typeof(CollectionsPage);
+        public MethodInfo TargetMethod => TargetType.GetMethod("draw");/*
+        public static void Postfix(CollectionsPage __instance, int currentTab, int index, ref string __result)
+        {
+            if (currentTab == 5 || currentTab == 6)
+                return;
+
+            BookcaseMod.logger.Alert(__result);
+            __result += "\nTest";
+        }*/
```
Comment (darkhax) [inline Bookcase/Patches/StardewValley/Menus/IClickableMenu/CollectionsPageDrawPatch.cs]: Probably should remove this lol

Comment (M3ales) [reply Bookcase/Patches/StardewValley/Menus/IClickableMenu/CollectionsPageDrawPatch.cs]: Done.

**Comment by darkhax** [comment]
Okay, reverted the harmony loader changes. When you resolve these conflicts I can merge this in.

**Pull request merged**